qupulse/_program/waveforms.py

```diff
@@ -226,7 +229,14 @@ def unsafe_sample(self,
         if output_array is None:
             output_array = np.empty_like(sample_times)
```

Comment:
```suggestion
            output_array = np.empty_like(sample_times)
            output_array.fill(np.NaN)
```
This PR fixes the example, but there might be other corner cases. This would make the problems more explicit.
(there are other places in the codebase where this applied as well)

Reply:
Probably a good idea. I added an `ALLOCATION_FUNCTION` for now to have the option to configure this with minimal overhead.